docs/dev/experimental-mode.adoc

```diff
@@ -19,74 +19,104 @@ Experimental mode can be enabled by setting odo preferences as well.
+Experimental mode is enabled, use at your own risk
+
+Validation
+ ✓  Checking Devfile compatibility [234445ns]
+ ✓  Validating Devfile [84112ns]
+
+Please use `odo push` command to create the component with source deployed
+```
+
+##### Creating and deleting component url:
+Create a devfile component url with `odo url create` and delete a devfile component url with `odo url delete`. This updates the component settings at `.odo/env/env.yaml` accordingly.
+
+When creating a component url, the ingress domain must be specified with the flag `--host` and a valid port must be specified with `--port`. The ports to be pushed, must be in the devfile.
+
+```
+$ odo url create myspring --port 8080 --host <ingress domain>
+ ✓  URL created for component: myspring, cluster host: <ingress domain>
+
+To create URL on the OpenShift Cluster, please use `odo push`
+```
+
+When deleting a component url, specify the url name to be deleted
 ```
+$ odo url delete myspring
+? Are you sure you want to delete the url myspring Yes
+ ✓  URL myspring removed from the env file
+
+To delete URL on the OpenShift Cluster, please use `odo push`
+```
+
+To push either of the url sub-command changes, execute `odo push`.
+
+##### Pushing to a devfile component:
+When the experimental mode is enabled, `odo push` creates a pod with all the devfile component containers and volumes. Push syncs the local project on disk to the pod container where devfile component has `mountSources: true`. The optional build and the mandatory run commands are executed in order.
+
+```
+$ odo push
+ •  Push devfile component myspring  ...
+
+Applying URL changes
+ ✓  URL myspring: http://myspring.<ingress domain> created
+ ✓  Waiting for component to start [65ms]
+ ✓  Checking files for pushing [10ms]
+ ✓  Syncing files to the component [1s]
+ ✓  Executing devbuild command "/artifacts/bin/build-container-full.sh" [1m]
+ ✓  Executing devrun command "/artifacts/bin/start-server.sh" [3s]
+ ✓  Push devfile component myspring [2m]
+ ✓  Changes successfully pushed to component
+```
+
+By default, `odo push` executes `devbuild` and `devrun` devfile commands. However, custom commands can be also be provided to `odo push` via flags `--build-command` & `--run-command`. `odo push` also provides a `--devfile` flag to execute push with a `devfile.yaml` in a custom path. These flags are only available in the experimental mode.
+
+##### Deleting a devfile component:
+Delete a devfile component with `odo delete`. This deletes all the Kubernetes resources created during `odo push`. Use the `-all` flag to delete the Kubernetes resources and the local config at `.odo/env/env.yaml`
+
+```
+$ odo delete myspring
+? Are you sure you want to delete the devfile component? Yes
+ ✓  Deleting devfile component myspring [339ms]
+ ✓  Successfully deleted component
+
+```
+
+#### Developing a devfile:
+
+The devfile registry can be accessed link:https://github.com/elsony/devfile-registry[here]. 
```

Comment:
We should highlight that we also pull devfiles from https://github.com/eclipse/che-devfile-registry, even if none of them are considered supported by odo at the moment